.github/workflows/test-env-deploy.yml

```diff
@@ -0,0 +1,63 @@
+name: TEST-ENV-DEPLOYMENT
+
+on:
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
+    branches: ["*"]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-qa-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-qa-${{ env.cache-name }}-
+            ${{ runner.os }}-qa-
+            ${{ runner.os }}-
+
+      - name: Install deps
+        run: |
+          npm ci
+
+      - name: Set API URI
+        env:
+          pull_request_body: ${{ github.event.pull_request.body }}
+          prefix: API_URI=
+          pattern: (http|https)://[a-zA-Z0-9.]+/graphql/?
```

Comment:
That's a URL not a URI 👮 

<img src='https://user-images.githubusercontent.com/6186720/86761944-9c417280-c046-11ea-837e-0dd3b4d27ade.png' width='300px'/>

Reply:
I was trying to be consistent with the variable naming in storefront itself

Reply:
But now, what if someone wants to say `/graphql/`? That would be a bug to not support it as we are expecting a URL and not a URI/URN.